### PR TITLE
Use netifaces to get network interface info

### PR DESCRIPTION
**Pull request by maxiberta**
The migration to `netifaces` will help with IPv6 support ([LP: #829379](https://bugs.launchpad.net/landscape-client/+bug/829379)).

### CODE REVIEW

File: landscape/client/broker/tests/public.ssl [hunk 1]
```diff
@@ -1,9 +1,9 @@
 -----BEGIN CERTIFICATE-----
```
Comment (sparkiegeek) [inline landscape/client/broker/tests/public.ssl]: I think you already fixed this and need to rebase?

File: landscape/client/manager/tests/test_usermanager.py [hunk 2]
```diff
@@ -483,17 +483,17 @@ def test_remove_user_event_leave_home(self):
...
-                self.assertEqual(len(messages), 3)
-                # Ignore the message created by plugin.run.
-                self.assertMessages([messages[2], messages[1]],
-                                    [{"timestamp": 0, "delete-users": ["jdoe"],
-                                      "type": "users", "operation-id": 39},
-                                     {"type": "operation-result",
-                                      "status": SUCCEEDED,
-                                      "operation-id": 39, "timestamp": 0,
-                                      "result-text": "remove_user succeeded"}])
+            messages = (
```
Comment (sparkiegeek) [inline landscape/client/manager/tests/test_usermanager.py]: Likewise for this chunk

File: landscape/lib/network.py [hunk 3]
```diff
@@ -10,129 +10,63 @@
...
-
-
-def get_active_interfaces(sock):
-    """Generator yields active network interface names.
-
-    @param sock: a socket instance.
-    """
-    max_interfaces = 128
+def get_active_interfaces():
+    for interface in netifaces.interfaces():
```
Comment (sparkiegeek) [inline landscape/lib/network.py]: Add a docstring please, calling out that this is a generator

File: landscape/lib/network.py [hunk 4]
```diff
@@ -10,129 +10,63 @@
...
-    return socket.inet_ntoa(fcntl.ioctl(
-        sock.fileno(),
-        SIOCGIFBRDADDR,
-        struct.pack("256s", interface[:15]))[20:24])
+    return ifaddresses[netifaces.AF_INET][0].get('broadcast', '0.0.0.0')
 
 
-def get_netmask(sock, interface):
+def get_netmask(ifaddresses):
     """Return the network mask associated to an interface.
```
Comment (sparkiegeek) [inline landscape/lib/network.py]: Docstring needs updates

File: landscape/client/broker/tests/public.ssl (hunk 1 above)
Comment (maxiberta) [reply landscape/client/broker/tests/public.ssl]: Right, gone by merging `master`.

File: landscape/client/manager/tests/test_usermanager.py (hunk 2 above)
Comment (maxiberta) [reply landscape/client/manager/tests/test_usermanager.py]: Right, gone by merging `master`.

File: landscape/lib/network.py (hunk 3 above)
Comment (maxiberta) [reply landscape/lib/network.py]: Updated docstring.

File: landscape/lib/network.py (hunk 4 above)
Comment (maxiberta) [reply landscape/lib/network.py]: Docstrings updated in all affected `get_*` functions.

File: debian/control [hunk 5]
```diff
@@ -40,6 +40,7 @@ Depends: ${python3:Depends}, ${misc:Depends}, ${extra:Depends},
          ${shlibs:Depends},
          landscape-common (= ${binary:Version}),
          python3-pycurl,
+         python3-netifaces,
```
Comment (simpoir) [inline debian/control]: landscape-sysinfo is actually part of landscape-common (shipped with ubuntu-server). That dependency should be in the other Package. (cloud-init pulls it so it likely will be already there, but still)

Comment (maxiberta) [reply debian/control]: Fixed, thanks!

**Pull request merged**